internal/doctl/mylib.go

```diff
@@ -0,0 +1,182 @@
+package mylib
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/digitalocean/godo"
+	"github.com/pkg/errors"
+)
+
+//DoctlDependencies interface for doctl dependent functions
+type DoctlDependencies interface {
+	IsAuthenticated(token string) error
+	GetCurrentDeployment(appID string) ([]byte, error)
+	RetrieveActiveDeploymentID(appID string) (string, error)
+	RetrieveActiveDeployment(deploymentID string, appID string) ([]byte, error)
+	UpdateAppPlatformAppSpec(appID string) error
+	CreateDeployments(appID string) error
+	RetrieveFromDigitalocean() ([]byte, error)
+	RetrieveAppID(appName string) (string, error)
+	IsDeployed(appID string) error
+	ReDeploy(input string, appName string) error
+}
+
+//DoctlServices is a struct for holding doctl dependent function interface
+type DoctlServices struct {
+	Dep DoctlDependencies
+}
+
+//UpdatedRepo used for parsing json object of changed repo
+type UpdatedRepo struct {
+	Name       string
+	Repository string
+	Tag        string
+}
+
+//IsAuthenticated used for user authentication
+func (d *DoctlServices) IsAuthenticated(token string) error {
+	val, err := exec.Command("sh", "-c", fmt.Sprintf("doctl auth init --access-token %s", token)).Output()
+	if err != nil {
+		return fmt.Errorf("unable to authenticate user: %s", val)
+	}
+	return nil
+}
+
+//GetCurrentDeployment takes appID as input and returns list of deployments (used to retrieve most recent deployment)
+func (d *DoctlServices) GetCurrentDeployment(appID string) ([]byte, error) {
```

Comment:
The command lists deployments other than the curernt deployment. The name can be updated to something more accurate like `ListDeployments` or the functionality can return the current deployment only.

Reply:
updated using ListDeployments